pkg/traceql/ast.go

```diff
@@ -265,7 +262,6 @@ func (o ScalarOperation) impliedType() StaticType {
 func (o ScalarOperation) extractConditions(request *FetchSpansRequest) {
 	o.LHS.extractConditions(request)
 	o.RHS.extractConditions(request)
-	request.AllConditions = false
```

Comment:
don't we need this to stay false? i think it's used when we do [pipeline level operations](https://github.com/grafana/tempo/blob/c75e9f88134bfb305b82d646ead2e49471501895/pkg/traceql/expr.y#L249-L258) like:

```
({ span.http.status_code = 200 } | count()) > ({ span.http.status_code = 500 } | count())
```

i know we don't support this yet all the way through the engine, but the parser does

Reply:
That is still AllConditions=false because of the aggregate.  I was thinking the scalar itself doesn't imply that it can't AllConditions.   For example `{ .foo="a"} | 3>2` is a scalar filter without an aggregate that is now AllConditions=true.  

Added a test case for both of those queries.  Is there any others we can think of?

Reply:
good call. re-reviewed the parser and i'm not thinking of any other great examples.